mlflow/protos/service.proto

```diff
@@ -978,9 +978,25 @@ message GetMetricHistory {
   // Name of the metric.
   optional string metric_key = 2 [(validate_required) = true];
 
+  // Token indicating the page of metric history to fetch
+  optional string page_token = 4;
+
+  // Maximum number of logged instances of a metric for a run to return per paged call.
+  // Backend servers may select a specific `max_results` value to alleviate memory pressure
+  // for extremely large queries. This pagination feature is not user-controlled and
+  // reassembly of the final metric history collection is handled directly by the client.
+  // Queries that do not specify this value will revert to non-paginated queries wherein all
+  // metric history values for a given metric within a run will be returned from a single query.
+  optional int64 max_results = 5;
+
   message Response {
     // All logged values for this metric.
     repeated Metric metrics = 1;
+
+    // Token that can be used to issue a query for the next page of metric history values.
+    // A missing token indicates that no additional metrics are available to fetch OR that
```

Comment:
The second case is a subset of the first, right? If so, we can remove the second case

Reply:
good point :)